apps/wallet/src/ui/app/staking/home/StakeAmount.tsx

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { memo } from 'react';
+
+import { Text } from '_app/shared/text';
+import { useFormatCoin } from '_hooks';
+
+//TODO unify StakeAmount and CoinBalance
+
+type Colors = 'gray-90' | 'success-dark' | 'gray-60' | 'steel-darker';
+
+interface StakeAmountProps {
+    balance: bigint;
+    type: string;
+    diffSymbol?: boolean;
+    color: Colors;
+    symbolColor: Colors | 'steel';
+    size: 'heading4' | 'body';
+}
+
+function StakeAmount({
```

Comment:
I don't love this component, I feel like instead of all of the props there probably are just a few known variations (from looking at usage):

- body / steel-darker / steel-darker
- heading4 / gray-90 / steel
- heading4 / gray-60 / gray-60 (although in the wallet figma this one actually looks like the symbol color is wrong and actually should be steel, and it looks like the earned is only gray-60 because the value is 0, which we can actually solve in the component itself by inspecting balance)

Assuming we cut this down to two, I think the prop interface really should be something closer to:

```ts
interface props {
  balance: bigint;
  variant: 'heading' | 'body'
}
```

I also don't think you need this to accept `type`, since it's statically known for staking.